common/src/hashing_domain.rs

```diff
@@ -30,10 +31,10 @@ pub struct HashingDomain {
 }
 
 /// Error type for the pipeline.
-#[derive(Debug, Error)]
+#[derive(Debug, Clone, Error, PartialEq, Eq, Serialize, Deserialize)]
 pub enum HashingDomainError {
     #[error("Two slices have different lengths")]
-    CopyFromSlice,
+    SlicesHaveDifferentLengths,
```

Comment:
Is this error produced somewhere besides `as_fixed_bytes`?
If yes, add `DigestTooShort` variant
If no, change to `DigestTooShort` variant.

In any event, this would be far more instructive:
```
    #[error("Cannot create a slice of length {0} from a digest of length {1}.")]
    DigestTooShort(usize, size),
```

(Could also leave the parameters out)